### ISSUE TITLE
template tags are exported into message

### ISSUE THREAD

**Issue by asarubbo** [rsyslog/rsyslog#5227]
I have ported my configuration from something like this:

```
$ModLoad imuxsock
$ModLoad imjournal 
$WorkDirectory /var/lib/rsyslog
$ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat
$IncludeConfig /etc/rsyslog.d/*.conf
$OmitLocalLogging on
$IMJournalStateFile imjournal.state

*.info;mail.none;authpriv.none;cron.none                /var/log/messages
authpriv.*                                              /var/log/secure
mail.*                                                  -/var/log/maillog
cron.*                                                  /var/log/cron
*.emerg                                                 :omusrmsg:*
uucp,news.crit                                          /var/log/spooler
local7.*                                                /var/log/boot.log

$DefaultNetstreamDriver gtls
$DefaultNetstreamDriverCAFile /etc/ssl/certs/ca-bundle.crt
$ActionSendStreamDriverMode 1
$ActionSendStreamDriverAuthMode anon

$ActionQueueType Disk
$ActionQueueFileName queue_file

template(name="rsyslog"
  type="list") {
    constant(value="{")
      constant(value="\"@timestamp\":\"")     property(name="timereported" dateFormat="rfc3339")
     constant(value="\",\"message\":\"")     property(name="msg" format="json")
      constant(value="\",\"programname\":\"") property(name="programname")
    constant(value="\"}\n")
}

module(load="imfile" PollingInterval="10")
input(type="imfile" File="my.path" Tag="my.tag" Facility="local3")
local3.* @@logstash.server:logstash.port;rsyslog
```
into something like:
```
global(workDirectory="/var/lib/rsyslog")
module(load="builtin:omfile" Template="RSYSLOG_TraditionalFileFormat")
module(load="imuxsock" SysSock.Use="off")
module(load="imjournal" StateFile="imjournal.state")

*.info;mail.none;authpriv.none;cron.none,local3.none  /var/log/messages
authpriv.*                                            /var/log/secure
mail.*                                                -/var/log/maillog
cron.*                                                /var/log/cron
*.emerg                                               :omusrmsg:*
uucp,news.crit                                        /var/log/spooler
local7.*                                              /var/log/boot.log

template(name="rsyslog" type="list" option.jsonf="on") {
       property(outname="@timestamp" name="timereported" dateFormat="rfc3339" format="jsonf")
       property(outname="programname" name="programname" format="jsonf")
       property(outname="message" name="msg" format="jsonf")
}

module(load="imfile" PollingInterval="10")
input(type="imfile" File="my.path" Tag="my.tag" Facility="local3")

local3.* action(type="omfwd" protocol="tcp" Target="logstash.server" Port="logstash.port" StreamDriverMode="1" StreamDriver="gtls" StreamDriverAuthMode="anon" queue.filename="queue_filexxx" queue.type="Disk" template="rsyslog")
```

The poor part, in my opinion was the template. I found it over the net and since worked pretty well I didn't focalize into it. However, by using the new template stile I have an issue related to the json exported to logstash.

The old style template has a syntax like this (in logstash):
```
"_source": {
    "@version": "1",
    "message": "1.2.3.4 - - [14/Sep/2023:14:09:56 +0200] \"GET / HTTP/1.1\" 403 722 \"-\" \"__UA__ python-requests/2.31.0\" \"-\"",
    "programname": "my.programname",
    "type": "rsyslog",
    "@timestamp": "2023-09-14T12:10:05.076Z",
    "port": 59114,
    "host": "my.host"
  },
```

while with the new template style I have:

```
 "_source": {
    "@version": "1",
    "message": "{\"@timestamp\":\"2023-09-14T13:08:25.042098+02:00\", \"programname\":\"my.programname\", 1.2.3.4 - - [14\\/Sep\\/2023:13:08:25 +0200] \\\"POST \\/svc\\/calc.jsp HTTP\\/2.0\\\" 200 2322 \\\"https:\\/\\/my.host\\/\\\" \\\"Mozilla" \\\"-\\\"}",
    "@timestamp": "2023-09-14T11:08:25.222Z",
    "port": 35608,
    "tags": [
      "_jsonparsefailure"
    ],
    "host": "my.host",
    "type": "rsyslog"
  },
  "fields": {
    "@timestamp": [
      "2023-09-14T11:08:25.222Z"
    ]
  },
```

So in short tags like `programname` and `@timestamp` are inglobated into `message` while previously was outside `message`. What I can do to get the old behavior?

**Comment by davidelang** [comment]
write a log out to a file using the template RSYSLOG_DebugFormat
that will show you all the variables and what is in them.

by default, imfile doesn't parse the messages, it assumes that the entire line 
is the message body, with the hostname being the current host and the timestamp 
being when the file was read. If instead you have a syslog formatted message in 
the file, everything ends up in the message body. There is an option for imfile 
to parse the message it reads from the file. I don't remember the option off the 
top of my head.

David Lang

**Comment by rgerhards** [comment]
We are sorry for the late reply. We assume it will no longer be useful for the original poster, but go through all issues in order not to miss feature request or bug reports. Also, the answer might be useful for other who come to this page via search engines. 

As of our analysis, there was most probably a bug in the version used when reporting this issues. However, that issue no longer exits in current builds (verified for 8.2512.0.l but probably applies to at least all 2025 builds). 

So I will close this issue. Should you experience a similar situation, please open a new issues. That will get much quicker attention than when a new report is buried in an old one - especially as the code base has changed.
